wagtail/models/__init__.py

```diff
@@ -3254,7 +3254,17 @@ def can_move_to(self, destination):
 
         # reject moves that are forbidden by subpage_types / parent_page_types rules
         # (these rules apply to superusers too)
-        if not self.page.specific.can_move_to(destination):
+        # – but only check this if the page is not already under the target parent.
+        # If it already is, then the user is just reordering the page, and we want
+        # to allow it even if the page currently violates the subpage_type /
+        # parent_page_type rules. This can happen if it was either created before
+        # the rules were specified, or it was done programmatically (e.g. to
+        # predefine a set of pages and disallow the creation of new subpages by
+        # setting subpage_types = []).
+
+        if (not self.page.is_child_of(destination)) and (
+            not self.page.specific.can_move_to(destination)
+        ):
```

Comment:
I originally added the check in `Page.can_move_to()` instead (see [`982873c` (#11944)](https://github.com/wagtail/wagtail/pull/11944/commits/982873c658077d98d7eb0985b74492b418d73177)), but ran into a few issues:

- `Page.can_move_to()` calls `Page.can_exist_under()` to check the `subpage_type`/`parent_page_type` rules.
  - The method name seems to suggest the check for "is child of" should be in `Page.can_exist_under()`, but unfortunately it's a classmethod, so we can't put it there.
- The `Page.can_move_to()` method is tested without using saved page instances, so the necessary tree properties for the `is_child_of` are unset:
  - https://github.com/wagtail/wagtail/blob/4616501fb9c04917bc748c8136ea8af5e1007209/wagtail/tests/test_page_model.py#L3123-L3136
  - Hence, I had to add additional `self.pk and parent.pk` guards which aren't really necessary in real cases.

Anyway, I'm not sure why we have both `PagePermissionTester.can_move_to()` and `Page.can_move_to()`...